test/core/rename/Rename_test.cpp

```diff
@@ -0,0 +1,324 @@
+            sparta::bind(root_node->getChildAs<sparta::Port>("dispatch.ports.out_"+unit_name+"_write"),
+                         root_node->getChildAs<sparta::Port>(unit_name + ".ports.in_sink_inst"));
+            sparta::bind(root_node->getChildAs<sparta::Port>("dispatch.ports.in_"+unit_name+"_credits"),
+                         root_node->getChildAs<sparta::Port>(unit_name+".ports.out_sink_credits"));
+            sparta::bind(root_node->getChildAs<sparta::Port>("rename.ports.in_rename_retire_ack"),
+                     root_node->getChildAs<sparta::Port>(unit_name+".ports.out_rob_retire_ack"));
+        }
+
+        // Bind the "LSU" SinkUnit to Dispatch
+        sparta::bind(root_node->getChildAs<sparta::Port>("dispatch.ports.out_lsu_write"),
+                     root_node->getChildAs<sparta::Port>("lsu.ports.in_sink_inst"));
+        sparta::bind(root_node->getChildAs<sparta::Port>("dispatch.ports.in_lsu_credits"),
+                     root_node->getChildAs<sparta::Port>("lsu.ports.out_sink_credits"));
+        sparta::bind(root_node->getChildAs<sparta::Port>("rename.ports.in_rename_retire_ack"),
+                     root_node->getChildAs<sparta::Port>("lsu.ports.out_rob_retire_ack"));
+    }
+
+    sparta::ResourceFactory<olympia::Decode,
+                                olympia::Decode::DecodeParameterSet> decode_fact;
+    olympia::DispatchFactory        dispatch_fact;
+    olympia::RenameFactory          rename_fact;
+    olympia::MavisFactoy            mavis_fact;
+    core_test::SourceUnitFactory    source_fact;
+    core_test::SinkUnitFactory      sink_fact;
+    rename_test::ROBSinkUnitFactory   rob_sink_fact;
+
+
+    std::vector<std::unique_ptr<sparta::TreeNode>> tns_to_delete_;
+    std::vector<sparta::ResourceTreeNode*>         exe_units_;
+
+    const std::string input_file_;
+    sparta::log::Tap test_tap_;
+};
+
+const char USAGE[] =
+    "Usage:\n"
+    "    \n"
+    "\n";
+
+sparta::app::DefaultValues DEFAULTS;
+
+// The main tester of Rename.  The test is encapsulated in the
+// parameter test_type of the Source unit.
+void runTest(int argc, char **argv)
+{
+    DEFAULTS.auto_summary_default = "off";
+    std::vector<std::string> datafiles;
+    std::string input_file;
+    bool enable_vector;
+
+    sparta::app::CommandLineSimulator cls(USAGE, DEFAULTS);
+    auto & app_opts = cls.getApplicationOptions();
+    app_opts.add_options()
+        ("output_file",
+         sparta::app::named_value<std::vector<std::string>>("output_file", &datafiles),
+         "Specifies the output file")
+        ("input-file",
+         sparta::app::named_value<std::string>("INPUT_FILE", &input_file)->default_value(""),
+         "Provide a JSON instruction stream",
+         "Provide a JSON file with instructions to run through Execute")
+        ("enable_vector",
+         sparta::app::named_value<bool>("enable_vector", &enable_vector)->default_value(false),
+         "Enable the experimental vector pipelines");
```

Comment:
😄 I believe you won't use this option

Reply:
Addressed in [fbe7264](https://github.com/riscv-software-src/riscv-perf-model/pull/40/commits/fbe7264a97b4aadea92369bdf93a544c3077551e)